domain/resource/service/resource.go

```diff
@@ -167,12 +190,34 @@ func (s *Service) GetApplicationResourceID(
 	if err := args.ApplicationID.Validate(); err != nil {
 		return "", errors.Errorf("application id: %w", err)
 	}
-	if args.Name == "" {
+	if strings.TrimSpace(args.Name) == "" {
```

Comment:
todo: Return to original code. We shouldn't add a restriction that does not exist when the charm is unmarshalled.

Here is the code in internal/charm which validates a resource name:
https://github.com/juju/juju/blob/b7a80a1457d89742973e989ab5b67c74be166dd6/internal/charm/resource/meta.go#L38-L41